Add plugin: MOC Link Helper

BogdanCodreanu opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/BogdanCodreanu/obsidian-moc-link-helper

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - **Plugin is desktop only**
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

BogdanCodreanu commented:
> Thank you for your submission, an automated scan of your plugin code's revealed the following issues:
> 
> ### Optional
> [[1](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/bb4bc066fdb1e26e610e99d9bb71e11142005302/src/utils/fileUtils.ts#L48)][[2](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/bb4bc066fdb1e26e610e99d9bb71e11142005302/src/utils/workspaceUtils.ts#L24)][[3](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/bb4bc066fdb1e26e610e99d9bb71e11142005302/src/utils/workspaceUtils.ts#L25)][[4](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/bb4bc066fdb1e26e610e99d9bb71e11142005302/src/main.ts#L88)]:Casting to `any` should be avoided as much as possible.
> 
> Do NOT open a new PR for re-validation.

I say these cases are fine for unnecessary complication of types as I am not defining any objects, just accessing some properties or stuff I don't care about. And besides, the stuff I'm accessing is null-checked

joethei commented:
[id: 'moc-link-helper:open-view',](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/main.ts#L41)
Don't prefix the command id with the plugin id, Obsidian will make sure that there are no conflicts with other plugins.

[.setName('Up Property Name')](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L52), [.setName('Parent Tag')](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L65), [Parent Notes](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L333), [Unlinked Note](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L344), ['Notes Included'](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L359), [label: 'Missing Notes',](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L361), [title="Selected Notes"](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L380)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[new Setting(containerEl)](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L82-L99)
There should not be a save button in settings, settings should be saved right after they are changed.

BogdanCodreanu commented:
> [id: 'moc-link-helper:open-view',](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/main.ts#L41) Don't prefix the command id with the plugin id, Obsidian will make sure that there are no conflicts with other plugins.
> 
> [.setName('Up Property Name')](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L52), [.setName('Parent Tag')](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L65), [Parent Notes](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L333), [Unlinked Note](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L344), ['Notes Included'](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L359), [label: 'Missing Notes',](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L361), [title="Selected Notes"](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/components/SideView.tsx#L380) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [new Setting(containerEl)](https://github.com/BogdanCodreanu/obsidian-moc-link-helper/blob/f32a3116b3ba15fe3bc009ffeaa1be6d7e481e09/src/settings/pluginSettings.ts#L82-L99) There should not be a save button in settings, settings should be saved right after they are changed.

Updated plugin's repo with the requested changes and changed the release files to contain the fixes. Also rebased.
